account_invoice_report_hide_line/models/account_invoice.py

```diff
@@ -13,5 +13,19 @@ class AccountInvoiceLine(models.Model):
     def _onchange_price_unit(self):
         if self.price_unit > 0.0:
             self.show_in_report = True
+        elif self.display_type:
+            self.show_in_report = True
         else:
             self.show_in_report = False
+            
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    show_special = fields.Boolean("Show Notes and Sections", default=True, 
```

Comment:
```suggestion
    show_note_section = fields.Boolean("Show Notes and Sections", default=True, 
```

May be more meaningful